src/domains/access.ml

```diff
@@ -80,11 +80,34 @@ let reset () =
 type acc_typ = [ `Type of CilType.Typ.t | `Struct of CilType.Compinfo.t * Offset.Unit.t ] [@@deriving eq, ord, hash]
 (** Old access type inferred from an expression. *)
 
+module MemoRoot =
+struct
+  include Printable.StdLeaf
+  type t = [`Var of CilType.Varinfo.t | `Type of CilType.Typ.t] [@@deriving eq, ord, hash]
```

Comment:
will deriving `eq` work here? Type is (potentially) recursive

Reply:
`CilType.Typ` exists precisely to take care of it. The recursion in `Cil.typ` structural comparison comes from `compinfo`s and `CilType.Compinfo` has an implementation that only compares `cid`s.